### ISSUE TITLE
Disabling animations breaks code depending on transitionend event

### ISSUE THREAD

**Issue by Mathiou04** [teamcapybara/capybara#2537]
## Meta
Capybara Version:
3.32.2
Driver Information (and browser if relevant):
3.142.7

## Expected Behavior
When setting `Capybara.disable_animation` to true, Javascript code depending on `transitionend` event executes.

## Actual Behavior
When setting `Capybara.disable_animation` to true, Javascript code depending on `transitionend` event is not executed.

## Steps to reproduce
Set `Capybara.disable_animation` to true.
Add this code to one of your page under test:
```
window.addEventListener("transitionend", () => {
    alert('Code executed');
});
```
The alert will not be displayed.

## Analysis

Today, when disabling animation, this CSS code is injected to all pages:
```
 transition: none !important;
 animation-duration: 0s !important;
 animation-delay: 0s !important; 
 ```
 The first line as the side effect of disabling transitions and therefore any code that depends on `transitionend` event will not be executed, potentially breaking pages that depend on it.
 
Is there a reason not to use the [following code](transition-delay: 0s !important;) instead of disabling all transitions?
```
transition-duration: 0s !important;
transition-delay: 0s !important;
```

**Comment by twalpole** [comment]
Setting 'disable_animation' does as the name says, it disables the animations, so yes it would also disable the transitionend events.  If you want those events either don't disable animations (have test specific settings in your app to speed them up) or monkeypatch Capybara::Server::AnimationDisabler::DISABLE_CSS_MARKUP_TEMPLATE to do whatever you prefer

**Comment by Mathiou04** [comment]
Hey @twalpole 
Thanks for the answer!

I wonder though why disabling the **transitions** (and then the `transitionend` event) but not the **animations** (the `animationend` event)?
It looks like it would be more consistent to have the same behaviour for the two no?

Also I had the impression that the intent of `disable_animation` was to speed up tests and make them less brittle, not really disabling all the code relying on animation/transitions?

_Note: I updated the above post the add the reference to the [web.dev article](https://web.dev/prefers-reduced-motion/#(bonus)-forcing-reduced-motion-on-all-websites) where I found the CSS code I referred to above_
